### ISSUE TITLE
Add support printing as an error stack trace of checkA11y method

### ISSUE THREAD

**Issue by NazarYermolenko** [abhinaba-ghosh/axe-playwright#18]
Example of test fail console log with Cucumber JS: 

``` ✔ When Campaigns page opened # step_defs/pages/slideroom/app/marketing/campaigns/campaigns.steps.ts:7
   ✖ Then user performs accessibility check # step_defs/pages/slideroom/app/general.steps.ts:15
       Error: Violations detected on page https://test-automation.slideroom-auto.net/#/marketing
           at Function.<anonymous> (/home/newfire/Desktop/work/qa-ui/framework/accessibility/AXE.ts:29:19)
           at step (/home/newfire/Desktop/work/qa-ui/framework/accessibility/AXE.ts:44:23)
           at Object.throw (/home/newfire/Desktop/work/qa-ui/framework/accessibility/AXE.ts:25:53)
           at rejected (/home/newfire/Desktop/work/qa-ui/framework/accessibility/AXE.ts:17:65)
   ✔ After # step_defs/setup.steps.ts:20
```
```
try {
            await checkA11y(page, elementContext, defaultOptions)
        } catch (err) {
            console.log("Error:", err)
            console.log(`^^^ Violations detected on page ${pageUrl} ^^^`)
            throw new Error(`Violations detected on page ${pageUrl}`)
        }
```

```
Error: AssertionError [ERR_ASSERTION]: 2 accessibility violations were detected
    at testResultDependsOnViolations (/home/newfire/Desktop/work/qa-ui/node_modules/axe-playwright/dist/index.js:61:26)
    at /home/newfire/Desktop/work/qa-ui/node_modules/axe-playwright/dist/index.js:52:5
    at Generator.next (<anonymous>)
    at fulfilled (/home/newfire/Desktop/work/qa-ui/node_modules/axe-playwright/dist/index.js:24:58) {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: 2,
  expected: 0,
  operator: 'strictEqual',
  [Symbol(originalCallSite)]: [Array],
  [Symbol(mutatedCallSite)]: [Array]
}```

**Comment by abhinaba-ghosh** [comment]
Hi @NazarYermolenko Lets make the idea clear. Currently even if the accessibility violations got detected, it passes the test. Do you want to fail the test if any of the violations got detected? In that case we can introduce a configuration param to handle the same. Thoughts?

**Comment by NazarYermolenko** [comment]
@abhinaba-ghosh actually, I couldn't remember the core of the issue, I resolved it in some way. Anyway, I'd like to get the ability to manipulate accessibility violations detailed report from the code. Could you extend the functionality of ax-playwright to get a detailed report in the code, for example, to convert a detailed report to CSV, or to parse it any other way?

**Comment by cipivanov** [comment]
> I'd like to get the ability to manipulate accessibility violations detailed report from the code.

That would be a really useful thing to have @NazarYermolenko. I would totally love if we could decouple the accessibility evaluation, reporting and violation result assertion.

Something that satisfies this interface will work great for me. I can simply parse the result myself, assert and fail a test run as I see fit.

```typescript
getViolations(page: Page, context?: ElementContext, options?: Options): Promise<Result[]>
```

Going further I see value in having the ability to inject our own, user-defined reporter.

```typescript
export interface Reporter {
  report(results: AxeResults): Promise<void>
}

// the current reporting capabilities are used as the default
printViolations(page: Page, reporter: Reporter, context?: ElementContext, options?: Options): Promise<Result[]>
````

Considering the library has the vast majority of it already implemented in some, should not be too massive of an undertaking.  Do you think it is a worthwhile undertaking @abhinaba-ghosh? I'm happy to get started with a small PR.

**Comment by abhinaba-ghosh** [comment]
@cipivanov Sounds exciting. Please go-ahead.

**Comment by abhinaba-ghosh** [comment]
Can you create necessary tickets for the feature and start working on them? This ticket will be a implementation of that feature. Will link those okce merged.

**Comment by cipivanov** [comment]
> Can you create necessary tickets for the feature and start working on them? This ticket will be a implementation of that feature. Will link those okce merged.

@abhinaba-ghosh not sure if I understand exactly what you mean by this, could you please clarify?

I've submitted a PR for what we discussed above. Let me know what do you think.

**Comment by abhinaba-ghosh** [comment]
@cipivanov changes are available in `v1.1.8`
Many thanks for the contribution.
